lib/rich-text.rb

```diff
@@ -15,17 +15,38 @@ def self.configure
 
 RichText.configure do |c|
   c.safe_mode = true
-  c.html_default_block_tag = 'p'
-  c.html_block_tags = {
-    firstheader: 'h1',
-    secondheader: 'h2',
-    thirdheader: 'h3',
-    list: ->(content, value) { %(<ol><li>#{value}</li></ol>) },
-    bullet: ->(content, value) { %(<ul><li>#{value}</li></ul>) }
-  }
-  c.html_inline_tags = {
-    bold: 'strong',
-    italic: 'em',
-    link: ->(content, value) { %(<a href="#{value}">#{content}</a>) }
+
+  c.html_default_block_format = :paragraph
+
+  c.html_formats = {
+    bold:           { type: :inline, tag: 'strong' },
+    br:             { type: :inline, tag: 'br' },
+    bullet:         { type: :block,  tag: 'li', parent: 'ul' },
+    div:            { type: :block,  tag: 'div' },
+    fifthheader:    { type: :block,  tag: 'h5' },
+    figure:         { type: :block,  tag: 'figure' },
+    firstheader:    { type: :block,  tag: 'h1' },
+    fourthheader:   { type: :block,  tag: 'h4' },
+    hr:             { type: :inline, tag: 'hr', block_format: false },
+    id:             { type: :block,  apply: ->(el, op){ el[:id] = op.attributes[:id] } },
+    image:          { type: :inline, tag: 'img', block_format: :figure, apply: ->(el, op){ el[:src] = op.value.dig(:image, :src) } },
+    ins:            { type: :inline, tag: 'ins' },
+    italic:         { type: :inline, tag: 'em' },
+    link:           { type: :inline, tag: 'a', apply: ->(el, op){ el[:href] = op.attributes[:link] } },
+    list:           { type: :block,  tag: 'li', parent: 'ol' },
+    paragraph:      { type: :block,  tag: 'p' },
+    position:       { type: :block,  apply: ->(el, op){
+      classes = (el[:class] || '').split(' ')
+      classes << op.attributes[:position]
+      el[:class] = classes.uniq.join(' ')
+      el
+    }},
+    secondheader:   { type: :block,  tag: 'h2' },
+    sixthheader:    { type: :block,  tag: 'h6' },
+    small:          { type: :inline, tag: 'small' },
+    strike:         { type: :inline, tag: 's' },
+    subscript:      { type: :inline, tag: 'sub' },
+    superscript:    { type: :inline, tag: 'sup' },
+    thirdheader:    { type: :block,  tag: 'h3' },
```

Comment:
or maybe give the option for them to define their own?

Reply:
I'm fine with paring it down; the formats themselves (and supporting tests) could move directly over into SBN. That said, seems like there should be _some_ simple formats so that the thing does something out of the box. Overriding the entire default collection with your own is really simple.

Reply:
I've moved the bulk of the definitions and their corresponding tests over the SBN in this branch here: https://github.com/voxmedia/sbn/compare/gm-richtext-formats?expand=1

What's left in this gem are a skeleton set of basic rules so that it does _something_ out of the box. Also, the tests for these formats have been pared down – tests now focus on specific  implementation details and less on general formats. I still need to write README documentation about configuring HTML rules, and will have that in by EOD tomorrow.